scripts/get_latest_release.sh

```diff
@@ -7,17 +7,31 @@
 REPO_TO_USE=$1
 BRANCH_TO_USE=$2
 
-LATEST_REL=$(curl --silent "https://api.github.com/repos/$REPO_TO_USE/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+BASIC_AUTH=""
+
+# If we find a github username and token, we use that.
+# In CI, these variables are available and useful to avoid rate limits which is
+# much more strict for unauthenticated requests.
+if [[ $GITHUB_USERNAME != "" && $GITHUB_TOKEN != "" ]];
+then
+	BASIC_AUTH="--user $GITHUB_USERNAME:$GITHUB_TOKEN"
+fi
+
+LATEST_REL=$(curl \
+	--silent \
+	$BASIC_AUTH \
+	"https://api.github.com/repos/$REPO_TO_USE/releases/latest" \
+	| grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
 
 # Check if this is a release branch
 THIS_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 
 if [[ $(echo "$THIS_BRANCH" | grep -c ^"$BRANCH_TO_USE") == 1 ]];
 then
     VERSION_REL=${THIS_BRANCH//$BRANCH_TO_USE/v}
-    REL_TO_USE=$(curl --silent "https://api.github.com/repos/$REPO_TO_USE/releases" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/' | sed -n "/$VERSION_REL/p" | sort -rV | head -n 1)
+    REL_TO_USE=$(curl --silent --user $BASIC_AUTH "https://api.github.com/repos/$REPO_TO_USE/releases" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/' | sed -n "/$VERSION_REL/p" | sort -rV | head -n 1)
 else
-    REL_TO_USE=$(echo "$THIS_BRANCH" | sed "s/\(.*\)/$LATEST_REL/")
+    REL_TO_USE=$LATEST_REL
```

Comment:
The sed command seemed to be redundantly used. I simplified it. Let me know if it was intentional somehow.

Reply:
You are correct - thank you for optimizing this.